runtime/js/40_http.js

```diff
@@ -144,6 +152,11 @@
           innerResp.headerList,
         ], respBody instanceof Uint8Array ? respBody : null);
       } catch (error) {
+        const connError = connErrors.get(httpConn);
```

Comment:
For this and the below line, I am not exactly 100% on this but I think (for normal Map at least) it is more performant to check membership and only get if true.
```javascript
if (error instanceof errors.BadResource && connErrors.has(httpConn)) {
  error = new connError.constructor(connErrors.get(httpConn).message);
}
```